### PR TITLE
Feature/migrate create object dict

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
Addresses #60 

Solution
========
Renamed and deleted attributes from the old ingredients
Created the "v1_v2_atrribute_changes" file to store merging changes 
Wrote tests for migration ingredients functions 

with @meganrm 

## Type of change
* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cellpack/autopack/Environment.py [hunk 1]
```diff
@@ -1949,7 +1961,7 @@ def getTotalNbObject(self, allIngredients, update_partner=False):
         totalNbIngr = 0
         for ingr in allIngredients:
             if ingr.type == "Grow":
-                totalNbIngr += int(ingr.left_to_place * (ingr.length / ingr.uLength))
+                totalNbIngr += int(ingr.left_to_place * (ingr.length / ingr.length))
```
Comment (mogres) [inline cellpack/autopack/Environment.py:1964]: I think this should still be `uLength`?

File: cellpack/autopack/ingredient/single_cube.py [hunk 2]
```diff
@@ -69,62 +58,49 @@ def __init__(
...
-        self.positions2 = positions2  # top right corner of cuboid
-        positions_ar = numpy.array(self.positions[0][0])
-        positions2_ar = numpy.array(self.positions2[0][0])
+        self.bb = bounds
 
+        self.lower_bound = bounds[0]  # bottom left corner of cuboid
+        self.upper_bound = bounds[1]  # top right corner of cuboid
+        lower_bound = numpy.array(self.lower_bound)
+        upper_bound = numpy.array(self.upper_bound)
+        self.edges = numpy.array([upper_bound[i] - upper_bound[i] for i in range(3)])
```
Comment (mogres) [inline cellpack/autopack/ingredient/single_cube.py:93]: Should this be `upper_bound[i] - lower_bound[i]`?

File: cellpack/autopack/loaders/recipe_loader.py [hunk 3]
```diff
@@ -123,6 +124,27 @@ def _migrate_version(recipe):
             recipe["bounding_box"] = recipe["options"]["boundingBox"]
         return recipe
 
+    @staticmethod
```
Comment (mogres) [inline cellpack/autopack/loaders/recipe_loader.py:127]: great addition!

File: cellpack/tests/test_representations.py [hunk 4]
```diff
@@ -68,3 +71,32 @@ def test_unpacked_sphere_tree():
...
+    representations = Representations(
+        mesh=test_mesh_obj, atomic=test_pdb_id_obj_with_offset
+    )
+    # if the representation used has a transform, should move the starting position
+    representations.set_active("atomic")
+    adjusted_position = representations.get_adjusted_position(
+        starting_position, np.eye(4)
+    )
+    for i in range(len(adjusted_position)):
+        assert adjusted_position[i] != starting_position[i]
```
Comment (mogres) [inline cellpack/tests/test_representations.py:94]: This only works if the translation occurs in all three dimensions. Could we modify this?
`adjusted_position[i] == starting_position[i] - test_pdb_id_ob_with_offset["transform"]["translate"][i]`

**Review by mogres — APPROVED** [review]
Some minor comments left at respective diffs, but otherwise everything looks good! Thanks for putting in all the variable name changes and the tests, these are super helpful.

File: cellpack/autopack/Environment.py (hunk 1 above)
Comment (meganrm) [reply cellpack/autopack/Environment.py:1964]: yeah these are changes from my recipe loader branch, you're right. I changed this to `unit_length` though. @rugeli I'll make this change

File: cellpack/tests/test_representations.py (hunk 4 above)
Comment (meganrm) [reply cellpack/tests/test_representations.py:94]: I guess I could also make a flag that just turns true if any of them don't match

File: cellpack/autopack/loaders/v1_v2_attribute_changes.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+v1_to_v2_name_map = {
+    "Type": "type",
+    "distExpression": "distance_expression",
+    "distFunction": "distance_function",
+    "isAttractor": "is_attractor",
+    "jitterMax": "max_jitter",
+    "nbJitter": "num_jitter",
+    "nbMol": "count",
+    "overwrite_distFunc": "	overwrite_dist_func",
```
Comment (meganrm) [inline cellpack/autopack/loaders/v1_v2_attribute_changes.py:9]: the spacing here looks weird, can you delete these extra spaces?

File: cellpack/autopack/loaders/v1_v2_attribute_changes.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+v1_to_v2_name_map = {
+    "Type": "type",
+    "distExpression": "distance_expression",
+    "distFunction": "distance_function",
+    "isAttractor": "is_attractor",
+    "jitterMax": "max_jitter",
+    "nbJitter": "num_jitter",
```
Comment (meganrm) [inline cellpack/autopack/loaders/v1_v2_attribute_changes.py:7]:
```suggestion
    "nbJitter": "jitter_attempts",
```

Comment (meganrm) [reply cellpack/autopack/loaders/v1_v2_attribute_changes.py:7]: we changed this to be more clearly named

File: cellpack/autopack/loaders/v1_v2_attribute_changes.py (hunk 5 above)
Comment (meganrm) [reply cellpack/autopack/loaders/v1_v2_attribute_changes.py:9]: the new name should also be `overwrite_distance_function`

**Pull request merged**